### ISSUE TITLE
Type number for search not true/incorrect results

### ISSUE THREAD

**Issue by lcdaugherty** [EBISPOT/ols4#624]
if I type in a search term here : https://www.ebi.ac.uk/ols4/search?q=hemoglobin
The result indicates the following types:
Type
Class (2749)
Individual (3)
Objectproperty (102)
Ontology (1)
Property (102)

However, if I then select one of the types eg: Ontology (there should be 1 but I get no results when selecting it

**Comment by haideriqbal** [comment]
@lcdaugherty thanks for reporting the bug. Examining the bug we've noticed that `ontology` shouldn't be in the list of term type filters as it doesn't make logical sense. So it will be fixed soon.

**Comment by hauschke** [comment]
I think it is really useful to have that filter. I am just a regular OLS user, but I oftentimes search for an ontology. There is a dedicated feature for it, but why shouldn't it be possible to use the search function, too?

**Comment by haideriqbal** [comment]
hi @hauschke, if I understood your question right it means you are searching for ontologies like "efo" in the search bar? I believe thats not the usecase for that search bar in the first place. We have a dedicated tab for ontologies where all the ontologies in the OLS can be searched. Please correct me if I'm wrong.

**Comment by jamesamcl** [comment]
I think this would be a good use case for the main search box as many users (including myself with my user hat on) come to OLS looking for a specific ontology before looking for terms.

Ontologies were always supposed to be included in the entity search.

**Comment by SArndt-TIB** [comment]
Thanks for discussing this! I also think using the search bar to search for ontologies is a valid use case and users tend to do it. Wasn't it also possible on OLS3 (at least [this call](https://www.ebi.ac.uk/ols4/api/search?q=*&page=1&includeimported=true&rows=50000&type=ontology) still works)? They may be used to it. Unfortunately, there do not seem to be included any entities of type ontology in the search results on the v2 API, e.g. [this call](https://www.ebi.ac.uk/ols4/api/v2/entities?search=*&size=10&page=0&facetFields=ontologyId+type&type=ontology). Interestingly, the ontology facet is populated.

Regarding the original reported issue - not sure, if this is really the bug that should be taken care of. Did you at some point in time have an ontology with "hemoglobin" in any of the search fields? The facet field might have been correct at that state of dataload. At least now, when I do <https://www.ebi.ac.uk/ols4/search?q=*> the facet shows the number "277" for ontology - that's the number of ontologies indexed on OLS right now. The counter seems to be right but the search results are not retrieved?

We also discussed this issue for TIB TS and one of my colleagues created a solution [here](https://github.com/TIBHannover/ols4/issues/137) that supports the searchers, I guess.

<img width="309" height="391" alt="Image" src="https://github.com/user-attachments/assets/9dc5d6c3-6b9a-4f58-8f4d-8d0404f539ca" />

<img width="352" height="208" alt="Image" src="https://github.com/user-attachments/assets/6a3c13dd-d17f-4c75-a8f5-e6d7c4519d73" />

**Comment by hauschke** [comment]
@haideriqbal

> if I understood your question right it means you are searching for ontologies like "efo" in the search bar? I believe thats not the usecase for that search bar in the first place. We have a dedicated tab for ontologies where all the ontologies in the OLS can be searched. Please correct me if I'm wrong.

No, you are right. But I agree with [SArndt-TIB](https://github.com/SArndt-TIB) that using the search bar to search for _any_ content of a system is a relevant use-case. I am a normal user of OLS, I use it not every 2-3 weeks. I know about some special features like the ontology tab, but after 2-3 weeks I forget about it again. Using the search seems to be the "normal" way to do it. It's just a UX thing.

**Comment by haideriqbal** [comment]
Thanks to all for the comments. I'll investigate and add the requested feature to include ontologies in our search results.

**Comment by matentzn** [comment]
FWIW I also agree with @hauschke and @SArndt-TIB 

> But I agree with @SArndt-TIB that using the search bar to search for any content of a system is a relevant use-case.

I think I would like that too, it would save me from my current workflow, which is 

1. bookmarking some core ontologies I am using in my browser
2. loading the bookmarked ontology (eg https://www.ebi.ac.uk/ols4/ontologies/hp)
3. changing the URL to the ontology I want (https://www.ebi.ac.uk/ols4/ontologies/chebi)

This is how I have been searching ontologies in OLS for years, but yeah, thats not great UX.

I know this issue is getting more and more off course, but I would like to be able to search for ontologies. 

[[[[[[[[[[[[[[### **please do not read this!** I would also like to use a simple search language like github to immediately restrict search to one or more ontologies like `ontology:hp,ontology:mondo`]]]]]]]]]]]]
